tt_metal/impl/dispatch/kernels/command_queue_producer.hpp

```diff
@@ -57,6 +71,49 @@ void issue_queue_pop_front(uint32_t cmd_size_B) {
     notify_host_of_issue_queue_read_pointer();
 }
 
+FORCE_INLINE
+void program_local_cb(uint32_t data_section_addr, uint32_t num_pages, uint32_t page_size, uint32_t cb_size) {
+    uint32_t cb_id = 0;
+    uint32_t fifo_addr = data_section_addr >> 4;
+    uint32_t fifo_limit = fifo_addr + (cb_size >> 4);
+    cb_interface[cb_id].fifo_limit = fifo_limit;  // to check if we need to wrap
+    cb_interface[cb_id].fifo_wr_ptr = fifo_addr;
+    cb_interface[cb_id].fifo_rd_ptr = fifo_addr;
+    cb_interface[cb_id].fifo_size = cb_size >> 4;
+    cb_interface[cb_id].tiles_acked = 0;
+    cb_interface[cb_id].tiles_received = 0;
+    cb_interface[cb_id].fifo_num_pages = num_pages;
```

Comment:
2 things: why does CQ have (mulitple?) cb init routines custom to itself (vs the common one in circular_buffer.h)?
also, the init of acked/received here can be done w/ a single write (see the other implementation) since they are both 16 bit and inited to 0.

Reply:
Pasting from conversation with @DrJessop 

```
one is a pure read interface
another is the interface between the producer and consumer
```

also can changes to do the ack/rcv with a single write be uplifted in follow up PR that focuses only on dispatch kernel additions/changes?